include/measurement_kit/nettests/ffi.hpp

```diff
@@ -204,215 +298,18 @@ class MK_NETTESTS_DEPRECATED BaseTest {
     // SharedPtr, this means that attempting to run more than one test
     // leads to an exception being thrown.
 
-    void run() { run_static(std::move(impl_)); }
-
-    void start(std::function<void()> &&fn) {
-        std::thread thread{[tip = std::move(impl_), fn = std::move(fn)]() {
-            run_static(std::move(tip));
-            fn();
-        }};
-        thread.detach();
+    void run() {
+        (void)cxx14::TaskRunner::global().run(std::move(impl_));
```

Comment:
Maybe, rather that ignoring the error, find a way to emit a warning?

Reply:
Also: worth thinking whether we want to return a `bool` or a bool-like instance with more info on the actual error that occurred (having more info can help to respond to bug reports etc.)